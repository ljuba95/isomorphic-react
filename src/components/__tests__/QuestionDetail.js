import {mapStateToProps, QuestionDetail} from "../QuestionDetail"
import renderer from 'react-test-renderer'
import React from 'react'

describe(`Question Detail Component`, () => {

  describe('The Container Element', () => {

    describe('mapStateToProps', () => {

      it('should map the state to props correctly', () => {
        const sampleQuestion = {
          question_id: 42,
          body: "hello"
        }
        const state = {
          questions: [sampleQuestion]
        }
        const ownProps = {
          question_id: 42
        }
        const componentState = mapStateToProps(state,ownProps)
        expect(componentState).toEqual(sampleQuestion)
      })

    })

  })

  describe('The display element', () => {

    it('should not regress', () => {
      const tree = renderer.create(
        <QuestionDetail
          title="Hi"
          body="there"
          answer_count={23}
          tags={['hehe','hee']}
        />
      )

      expect(tree.toJSON()).toMatchSnapshot()
    })

  })


})