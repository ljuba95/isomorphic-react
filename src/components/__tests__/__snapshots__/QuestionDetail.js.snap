// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Detail Component The display element should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    Hi
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          hehe 
        </code>
        <code>
          hee 
        </code>
      </div>
    </div>
    <div>
      <p>
        there
      </p>
    </div>
    <div>
      23
       Answers
    </div>
  </div>
</div>
`;
